### PR TITLE
Add two sets of fxns, one for buckets, one for files

### PR DESCRIPTION
**Pull request by sckott**
@seankross  This PR focuses on S3 buckets and objects. The code works, but the goal I have here is to get your feedback on the high level approach of a suite of functions for buckets and a suite of functions for "files", where files is a synonym for object within a bucket. 

My idea is that many users will most often use the file fxns, and only time to time use the bucket fxns - so it probably makes sense to put effort into making the file fxns easy to use and do things automagically for users, e.g., in a file upload fxn, they could name the bucket, and if the bucket doesn't exist we could create it for them - or at least a usethis style informative message with the code to create a bucket. 

The other issue I want to discuss in this PR is about making it easy to find functions. I've taken the approach here of prefixing functions with `aws_`. I've used this approach in other pkgs, and I know other packages have as well, e.g., `fs`. E.g., take the function `aws_file_upload`. If it was just `file_upload` it'd be hard for the user to remember where that fxn came from unless they namespaced with the pkg name: `sixtyfour::file_upload`.

### CODE REVIEW

**Review by seankross — APPROVED** [review]
I had some time to mess around with this, looks and feels great overall. Biggest piece of feedback is that eventually `aws_bucket_list_objects`, `aws_file_attr`, and similar functions should return data frames, and we should get really opinionated what the columns of those data frames should be.

File: R/bucket.R [hunk 1]
```diff
@@ -1,62 +1,57 @@
...
+#'
+#' @export
+#' @param bucket (character) bucket name. required
+#' @param ... named parameters passed on to [list_objects](https://www.paws-r-sdk.com/docs/s3_create_bucket/)
+#' @examples \dontrun{
+#' aws_bucket_create(bucket="s64-test-2")
+#' }
+aws_bucket_create <- function(bucket, ...) {
+  env64$s3$create_bucket(Bucket = bucket,
+    CreateBucketConfiguration = list(LocationConstraint = "us-west-2"), ...)
```
[the remark this answers is not in the document]
Comment (sckott) [reply R/bucket.R:29]: Yes! Probably only the env var approach. For consistency for all the aws config options, it'd be good to keep those out of any function arguments as folks may put those in public places in code more easily than env vars

**Comment by sckott** [comment]
> ... similar functions should return data frames, and we should get really opinionated what the columns of those data frames should be.

Thanks. Yeah agree on that.

**Comment by sckott** [comment]
How do you feel about the top level function prefix of `aws_`?

**Comment by seankross** [comment]
> How do you feel about the top level function prefix of aws_?

I like it because it's short, in general I like this approach.

**Comment by sckott** [comment]
Cool, thanks

**Pull request merged**